### ISSUE TITLE
[Bug]: Running multiple inferences with the same compiled model and input produces different outputs

### ISSUE THREAD

**Issue by Mohamed-Ashraf273** [openvinotoolkit/openvino#32045]
### OpenVINO Version

openvino version: 2025.3.0

### Operating System

Ubuntu 18.04 (LTS)

### Device used for inference

CPU

### Framework

None

### Model used

Gemma

### Issue description

Running multiple inferences with the same compiled model and input produces different outputs.

### Relevant log output

```shell

```

### Issue submission checklist

- [x] I'm reporting an issue. It's not a question.
- [x] I checked the problem with the documentation, FAQ, open issues, Stack Overflow, etc., and have not found a solution.
- [x] There is reproducer code and related data files such as images, videos, models, etc.

**Comment by abheesht17** [comment]
Are you seeing this difference in numerics on top of this PR: https://github.com/keras-team/keras-hub/pull/2389, or is this reproducible on `master` as well?

I can see the test failure for [OpenVino](https://github.com/keras-team/keras-hub/actions/runs/17601783278/job/50004977106?pr=2389) here; we haven't merged it yet.

**Comment by Mohamed-Ashraf273** [comment]
@abheesht17 
No, it’s not caused by the PR itself. The issue is related to a bug in OpenVINO, re-running multiple inferences with the same compiled model produces different outputs (specifically on the test scenario defined for gemma, not for a real gemma inference, and not in any other models).

I noticed that OpenVINO currently has 4 failing tests on your PR. I’ve already fixed one on the Keras side (about score logits) here: [21664](https://github.com/keras-team/keras/pull/21664), but the remaining failures share the same root cause, which led me to discover the issue in OpenVINO.

I’m still not sure why it specifically affects Gemma, but to be clear, it’s not introduced by this PR.

**Comment by rkazants** [comment]
Hi @Mohamed-Ashraf273,

Please attach IR of the model and a simple script to reproduce this issue using OpenVINO API. Just avoid keras specific:)

It is possible that from one inference to another inference results can be different, if the model is stateful, for example.

Best regards,
Roman

**Comment by Mohamed-Ashraf273** [comment]
@rkazants 
Here is my reproducer:
```python
import openvino as ov
import numpy as np

core = ov.Core()
ov_model = core.read_model("model.xml")
ov_compiled_model = core.compile_model(ov_model, "CPU")

arr = np.random.rand(2, 8).astype(np.float32)

result1 = ov_compiled_model([arr])[0]
result2 = ov_compiled_model([arr])[0]

are_consistent = np.allclose(result1, result2)
max_difference = np.max(np.abs(result1 - result2))

print(f"Inference results are consistent: {are_consistent}")
print(f"Maximum difference between results: {max_difference}")
print(f"Result 1 shape: {result1.shape}, Result 2 shape: {result2.shape}")
```
output:
```
Inference results are consistent: False
Maximum difference between results: 3.2739830017089844
Result 1 shape: (2, 8, 11), Result 2 shape: (2, 8, 11)
```
IR link: https://drive.google.com/drive/folders/16iGXbcUVe5FikiWh2YlGbuNTgUASyMoE?usp=sharing

**Comment by rkazants** [comment]
Hi @Mohamed-Ashraf273,

Your model has RandomUniform operation in the graph that has a state. This is probably a reason of this difference.

Please elaborate why this bug is important.

Best regards,
Roman

**Comment by Mohamed-Ashraf273** [comment]
@rkazants
This issue specifically occurs with the `Gemma` model in `keras_hub`, while other models function correctly.
With the same model, it is used in this test, which leads to failures in this PR: https://github.com/keras-team/keras-hub/pull/2389

Also, `OpenVINO` fails due to a numeric mismatch for the same reason in this test (for Gemma):
https://github.com/keras-team/keras-hub/blob/master/keras_hub/src/models/gemma/gemma_causal_lm_test.py#L55

Since the same test passes on all other backends (TensorFlow, PyTorch, and JAX), it should also pass on OpenVINO.

**Comment by rkazants** [comment]
> [@rkazants](https://github.com/rkazants) This issue specifically occurs with the `Gemma` model in `keras_hub`, while other models function correctly. With the same model, it is used in this test, which leads to failures in this PR: [keras-team/keras-hub#2389](https://github.com/keras-team/keras-hub/pull/2389)
> 
> Also, `OpenVINO` fails due to a numeric mismatch for the same reason in this test (for Gemma): https://github.com/keras-team/keras-hub/blob/master/keras_hub/src/models/gemma/gemma_causal_lm_test.py#L55
> 
> Since the same test passes on all other backends (TensorFlow, PyTorch, and JAX), it should also pass on OpenVINO.

@Mohamed-Ashraf273, try to fix this issue on Keras side. You just need to fix conversion logic that uses RandomUniform operation. Set both seeds during creation of RandomUniform.

Best regards,
Roman

**Comment by tanishkadhaybar9** [comment]
Hello, myself Tanishka. I am a full-stack web developer, and I am very interested in contributing to your organization. I would love to be a part of your community and work on your projects. Could you assign me this task?

**Comment by Mohamed-Ashraf273** [comment]
@tanishkadhaybar9 
Just wanted to let you know that this issue has already been resolved. Thanks!
